lib/tasks/whitehall_importer.rb

```diff
@@ -149,13 +151,25 @@ def initial_status(whitehall_edition, revision)
       )
     end
 
+    def set_current_status(whitehall_edition, edition)
+      edition.status = Status.new(
+        state: whitehall_edition["state"],
+        revision_at_creation: edition.revision,
+      )
+
+      edition.save!
+    end
+
+    def set_published_state_first?(state)
```

Comment:
I know you were struggling with the name of this, how about just calling a spade a spade e.g `withdrawn?`. I think it's clear enough from the code by having `initial_status` and `set_current_status` that if the edition is withdrawn there is two statuses?

Reply:
I thought there might be more statuses that need to know about the last published state, but I can see this it not the case, so I'll rename, or perhaps remove, this method.